express-notes-api/index.js

```diff
@@ -0,0 +1,97 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+
+app.use(express.json());
+
+// GET
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  const notes = data.notes;
+  for (const property in notes) {
+    arr.push(notes[property]);
+  }
+  res.send(arr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const notes = data.notes;
+  if (isNaN(req.params.id) || req.params.id < 0) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  }
+  if (req.params.id === undefined) {
+    return res.status(404).send({ error: `cannot find note with id ${req.params.id}` });
+  }
+  for (const property in notes) {
+    if (property === req.params.id) {
+      res.status(200).send(notes[property]);
+    }
+  }
```

Comment:
You don't need a loop here, and if the id doesn't exist a response will never be sent to the client.

You can do something like this instead:

```JAVASCRIPT
if (notes[req.params.id]) {
  // Note exists, send not to client
} else {
  // Note doe snot exist, send 404 with error message
}
```